### DISCUSSION TITLE
RaspberryMatic mit HA + HM-MOD-RPI-PCB oder HmIP-RFUSB

### DISCUSSION THREAD

**Discussion by MrJanD** [OpenCCU/OpenCCU#1957] [🇩🇪 Fragen&Antworten] [answered]
Hallo zusammen,

erst einmal ein dickes Dankeschön für die Mühe und Arbeit die mit und an Homematic geleistet wird! Mit meinem HM-MOD-RPI-PCB und FHEM habe ich nie wirklich Probleme gehabt und es war immer ein sicher und gut funktionierendes System. 
Trotzdem habe ich mir mal vorgenommen, mich ein wenig in Home Assistant einzuarbeiten und mein Smart Home schrittweise zu migrieren. Nach ersten Erfolgen wie einfach und gut ESPHome Geräte mit Home Assistant zusammen arbeiten wollte ich mich einmal an Homematic mit meinen HM-CC-RT-DN heranwagen. Hier bin ich aber relativ schnell auf Probleme gestoßen. 
Wie im Titel beschrieben habe ich einen HM-MOD-RPI-PCB (bereits viele Jahre erfolgreich mit FHEM betrieben) und noch einen HmIP-RFUSB. Beide Module sind zwar nicht zeitgleich an einem System angeschlossen, jedoch habe ich mit beiden Modulen genau die gleichen Probleme: ich kann keine Geräte einlernen. Angehängte Logs und Screenshots beziehen sich ausschließlich auf den HM-MOD-RPI-PCB, wenn nötig kann ich gerne noch welche mit dem HmIP-RFUSB nachliefern, im Prinzip ist das Verhalten aber das selbe.

Mein Problem: Ich habe meine Home Assistant Installation anhand der Beschreibung auf https://github.com/jens-maus/RaspberryMatic/wiki/Installation-HomeAssistant und mit Hilfe des verlinkten Videos bereits 2x vorgenommen. Die Installation verläuft noch recht problemlos, jedoch kommen die Probleme sobald ich Geräte einlernen möchte. Mein Duty Cycle und Carrier Sense sieht mit dem HM-MOD-RPI-PCB so aus:
![image](https://user-images.githubusercontent.com/40394719/192144250-1758d1c1-a88d-4c7b-bc13-1da317900055.png)

Mit dem HmIP-RFUSB liegt der Carrier Sense bei 0%. Das sind aber auch schon die einzigen Unterschiede, welche ich festgestellt habe.
Ich vermute, dass das ganze irgendwie mit dem rfd zusammenhängt, da die Home Assistant Logs mit beiden Modulen wie folgt aussehen.

> Mounting /data as /usr/local (Home Assistant Add-On): OK
Starting watchdog...
Identifying host system: oci, OK
Initializing RTC Clock: no hardware found
Running sysctl: OK
Checking for Factory Reset: not required
Checking for Backup Restore: not required
Initializing System: OK
Setup ca-certificates: OK
Starting logging: OK
Init onboard LEDs: init, OK
Starting irqbalance: OK
Starting iptables: OK
Starting network: eth0: link up, fixed, firewall, inet up, 172.30.33.0, OK
Identifying Homematic RF-Hardware: ....HmRF: HM-MOD-RPI-PCB/GPIO@fe201000.serial, HmIP: HM-MOD-RPI-PCB/GPIO@fe201000.serial, OK
Updating Homematic RF-Hardware: HM-MOD-RPI-PCB: 2.8.6, not necessary, OK
Starting hs485dLoader: disabled
Starting xinetd: OK
Starting eq3configd: OK
Starting lighttpd: OK
Starting ser2net: disabled
Starting ssdpd: OK
Starting ha-proxy: OK
Starting NUT services: disabled
Initializing Third-Party Addons: OK
Starting LGWFirmwareUpdate: ...OK
Setting LAN Gateway keys: OK
Starting hs485d: disabled
Starting multimacd: .OK
Starting rfd: ....................ERROR
Starting HMIPServer: ............OK
Starting ReGaHss: .OK
Starting CloudMatic: OK
Starting NeoServer: disabled
Starting Third-Party Addons: OK
find: /usr/local/etc/monit*.cfg: No such file or directory
Starting crond: OK
Setup onboard LEDs: booted, OK
Finished Boot: 3.65.8.20220831 (raspmatic_oci_arm64)

In den RaspberryMatic Alarm Messages sehe ich ebenfalls den Fehler
![image](https://user-images.githubusercontent.com/40394719/192144595-6b330a84-089d-4e3b-97af-a57ab2d51633.png)

Das RaspberryMatic Log File sieht ebenfalls nicht gut aus:

> ***** messages *****
Sep 25 14:52:59 de838cd8-raspberrymatic syslog.info syslogd started: BusyBox v1.35.0
Sep 25 14:52:59 de838cd8-raspberrymatic daemon.debug /usr/sbin/irqbalance: IRQ vgic(9) guessed as class 0
Sep 25 14:52:59 de838cd8-raspberrymatic daemon.debug /usr/sbin/irqbalance: IRQ arch_timer(11) guessed as class 0
Sep 25 14:52:59 de838cd8-raspberrymatic daemon.debug /usr/sbin/irqbalance: IRQ vtimer(12) guessed as class 0
Sep 25 14:52:59 de838cd8-raspberrymatic daemon.debug /usr/sbin/irqbalance: IRQ fe00b880.mailbox(14) guessed as class 0
Sep 25 14:52:59 de838cd8-raspberrymatic daemon.debug /usr/sbin/irqbalance: IRQ (17) guessed as class 0
Sep 25 14:52:59 de838cd8-raspberrymatic daemon.debug /usr/sbin/irqbalance: IRQ ttyS0(18) guessed as class 0
Sep 25 14:52:59 de838cd8-raspberrymatic daemon.debug /usr/sbin/irqbalance: IRQ fe804000.i2c(19) guessed as class 0
Sep 25 14:52:59 de838cd8-raspberrymatic daemon.debug /usr/sbin/irqbalance: IRQ IRQ(22) guessed as class 0
Sep 25 14:52:59 de838cd8-raspberrymatic daemon.debug /usr/sbin/irqbalance: IRQ doorbell(29) guessed as class 0
Sep 25 14:52:59 de838cd8-raspberrymatic daemon.debug /usr/sbin/irqbalance: IRQ mmc0(30) guessed as class 0
Sep 25 14:52:59 de838cd8-raspberrymatic daemon.debug /usr/sbin/irqbalance: IRQ arm-pmu(32) guessed as class 0
Sep 25 14:52:59 de838cd8-raspberrymatic daemon.debug /usr/sbin/irqbalance: IRQ arm-pmu(33) guessed as class 0
Sep 25 14:52:59 de838cd8-raspberrymatic daemon.debug /usr/sbin/irqbalance: IRQ arm-pmu(34) guessed as class 0
Sep 25 14:52:59 de838cd8-raspberrymatic daemon.debug /usr/sbin/irqbalance: IRQ arm-pmu(35) guessed as class 0
Sep 25 14:52:59 de838cd8-raspberrymatic daemon.debug /usr/sbin/irqbalance: IRQ eth0(38) guessed as class 5
Sep 25 14:52:59 de838cd8-raspberrymatic daemon.debug /usr/sbin/irqbalance: IRQ eth0(39) guessed as class 5
Sep 25 14:52:59 de838cd8-raspberrymatic daemon.debug /usr/sbin/irqbalance: IRQ feb10000.codec(44) guessed as class 0
Sep 25 14:52:59 de838cd8-raspberrymatic daemon.debug /usr/sbin/irqbalance: IRQ v3d(45) guessed as class 0
Sep 25 14:52:59 de838cd8-raspberrymatic daemon.debug /usr/sbin/irqbalance: IRQ xhci_hcd(47) guessed as class 0
Sep 25 14:53:00 de838cd8-raspberrymatic user.info firewall: configuration set
Sep 25 14:53:09 de838cd8-raspberrymatic daemon.err xinetd[955]: Unable to read included directory: /etc/config/xinetd.d [file=/etc/xinetd.conf] [line=14]
Sep 25 14:53:09 de838cd8-raspberrymatic daemon.crit xinetd[955]: 955 {init_services} no services. Exiting...
Sep 25 14:53:09 de838cd8-raspberrymatic user.info root: Updating RF Lan Gateway Coprocessor Firmware
Sep 25 14:53:09 de838cd8-raspberrymatic user.debug update-coprocessor: firmware filename is: coprocessor_update_hm_only.eq3
Sep 25 14:53:09 de838cd8-raspberrymatic user.info root: Updating RF Lan Gateway Firmware
Sep 25 14:53:09 de838cd8-raspberrymatic user.info update-lgw-firmware: No gateway found in config file /etc/config/rfd.conf
Sep 25 14:53:16 de838cd8-raspberrymatic user.err rfd: CCU2CommController::improvedInit() - Identify response string not handled: 
Sep 25 14:53:21 de838cd8-raspberrymatic user.err rfd: () CCU2CommController::setCSMACAEnabled(): Error disabling CSMA/CA.
Sep 25 14:53:21 de838cd8-raspberrymatic user.err rfd: No BidCoS-Interface available
Sep 25 14:54:23 de838cd8-raspberrymatic daemon.info : starting pid 1336, tty '': '/bin/mv /tmp/boot.log /var/log/boot.log'
Sep 25 14:54:23 de838cd8-raspberrymatic daemon.info : starting pid 1337, tty '/dev/null': '/usr/bin/monit -Ic /etc/monitrc'
Sep 25 14:54:23 de838cd8-raspberrymatic user.info monit[1337]: Starting Monit 5.30.0 daemon with http interface at /var/run/monit.sock
Sep 25 14:54:23 de838cd8-raspberrymatic user.info monit[1337]: 'de838cd8-raspberrymatic' Monit 5.30.0 started
Sep 25 14:54:28 de838cd8-raspberrymatic local0.err ReGaHss: ERROR: XmlRpc: Error in XmlRpcClient::writeRequest: write error (error 111). [error():iseXmlRpc.h:281]
Sep 25 14:54:28 de838cd8-raspberrymatic local0.warn ReGaHss: WARNING: XMLRPC 'init': rpcClient.execute() failed (url: xmlrpc_bin://127.0.0.1:32001, params: {"xmlrpc_bin://127.0.0.1:31999","1007"}, result: nil) [CallXmlrpcMethod():iseXmlRpc.cpp:2602]
Sep 25 14:54:28 de838cd8-raspberrymatic local0.err ReGaHss: ERROR: XMLRPC 'init' call failed (interface: 1007, params: {"xmlrpc_bin://127.0.0.1:31999","1007"}) [CallInit():iseXmlRpc.cpp:1204]
Sep 25 14:54:28 de838cd8-raspberrymatic local0.err ReGaHss: ERROR: failed CallInit() for interface=BidCos-RF [ThreadFunction():iseRTHss.cpp:163]
Sep 25 14:54:39 de838cd8-raspberrymatic user.err monit[1337]: 'udevdEnabled' status failed (0) -- no output
Sep 25 14:54:40 de838cd8-raspberrymatic user.err monit[1337]: 'sshdEnabled' status failed (1) -- no output
Sep 25 14:54:40 de838cd8-raspberrymatic user.err monit[1337]: 'hs485dEnabled' status failed (2) -- grep: /var/etc/hs485d.conf: No such file or directory
Sep 25 14:54:40 de838cd8-raspberrymatic user.err monit[1337]: 'hmlangwEnabled' status failed (1) -- no output
Sep 25 14:54:40 de838cd8-raspberrymatic user.warn monit[1337]: 'hasUSB' status failed (1) -- no output
Sep 25 14:54:41 de838cd8-raspberrymatic user.err monit[1337]: 'hb_rf_eth-CheckEnabled' status failed (1) -- no output
Sep 25 14:54:41 de838cd8-raspberrymatic user.err monit[1337]: 'coProcessorCheck' status failed (1) -- no output
Sep 25 14:54:41 de838cd8-raspberrymatic user.warn monit[1337]: 'rpi4usb3Check' status failed (1) -- no output
Sep 25 14:54:41 de838cd8-raspberrymatic user.err monit[1337]: 'tailscaleEnabled' status failed (1) -- no output
Sep 25 14:54:41 de838cd8-raspberrymatic user.err monit[1337]: 'rfd' process is not running
Sep 25 14:54:41 de838cd8-raspberrymatic user.info monit[1337]: 'rfd' trying to restart
Sep 25 14:54:41 de838cd8-raspberrymatic user.info monit[1337]: 'rfd' restart: '/etc/init.d/S61rfd restart'
Sep 25 14:54:43 de838cd8-raspberrymatic user.err monit[1337]: Lookup for '/media/usb1' filesystem failed  -- not found in /proc/self/mounts
Sep 25 14:54:43 de838cd8-raspberrymatic user.err monit[1337]: Filesystem '/media/usb1' not mounted
Sep 25 14:54:43 de838cd8-raspberrymatic user.err monit[1337]: 'usb1' unable to read filesystem '/media/usb1' state
Sep 25 14:54:43 de838cd8-raspberrymatic user.info monit[1337]: 'usb1' trying to restart
Sep 25 14:54:59 de838cd8-raspberrymatic user.warn monit[1337]: 'hasUSB' status failed (1) -- no output
Sep 25 14:55:00 de838cd8-raspberrymatic user.warn monit[1337]: 'rpi4usb3Check' status failed (1) -- no output
Sep 25 14:55:00 de838cd8-raspberrymatic user.err monit[1337]: 'rfd' service restarted 1 times within 1 cycles(s) - exec
Sep 25 14:55:00 de838cd8-raspberrymatic user.info monit[1337]: 'rfd' exec: '/bin/triggerAlarm.tcl rfd restarted WatchDog: rfd-restart true'
Sep 25 14:55:00 de838cd8-raspberrymatic user.info monit[1337]: 'rfd' process is running with pid 1460
Sep 25 14:55:00 de838cd8-raspberrymatic user.info monit[1337]: 'rfd' process is running after previous restart timeout (manually recovered?)
Sep 25 14:55:00 de838cd8-raspberrymatic user.err monit[1337]: Filesystem '/media/usb1' not mounted
Sep 25 14:55:00 de838cd8-raspberrymatic user.err monit[1337]: 'usb1' unable to read filesystem '/media/usb1' state
Sep 25 14:55:00 de838cd8-raspberrymatic user.info monit[1337]: 'usb1' trying to restart
Sep 25 14:55:16 de838cd8-raspberrymatic user.warn monit[1337]: 'hasUSB' status failed (1) -- no output
Sep 25 14:55:16 de838cd8-raspberrymatic user.err monit[1337]: 'rpi4usb3Check' status failed (1) -- no output
Sep 25 14:55:16 de838cd8-raspberrymatic user.err monit[1337]: Filesystem '/media/usb1' not mounted
Sep 25 14:55:16 de838cd8-raspberrymatic user.err monit[1337]: 'usb1' unable to read filesystem '/media/usb1' state
Sep 25 14:55:16 de838cd8-raspberrymatic user.info monit[1337]: 'usb1' trying to restart
Sep 25 14:55:32 de838cd8-raspberrymatic user.warn monit[1337]: 'hasUSB' status failed (1) -- no output
Sep 25 14:55:33 de838cd8-raspberrymatic user.err monit[1337]: Filesystem '/media/usb1' not mounted
Sep 25 14:55:33 de838cd8-raspberrymatic user.err monit[1337]: 'usb1' unable to read filesystem '/media/usb1' state
Sep 25 14:55:33 de838cd8-raspberrymatic user.info monit[1337]: 'usb1' trying to restart
Sep 25 14:55:49 de838cd8-raspberrymatic user.err monit[1337]: 'hasUSB' status failed (1) -- no output
***** hmserver.log *****
2022-09-25 14:54:05,064 de.eq3.cbcs.server.core.vertx.KeyServerWorker ERROR [vert.x-worker-thread-3] Missing key server configuration parameter (Network.Key) for  mode: KEYSERVER_LOCAL 
2022-09-25 14:54:16,495 de.eq3.ccu.bidcos.dispatcher.BidCosRpcDispatcher ERROR [Thread-2] BidCosCache could not be initialized 
2022-09-25 14:54:16,496 de.eq3.ccu.bidcos.dispatcher.BidCosRpcDispatcher ERROR [Thread-2] java.net.ConnectException: Connection refused (Connection refused) 
2022-09-25 14:54:18,069 de.eq3.ccu.bidcos.dispatcher.BidCosRpcDispatcher ERROR [Timer-2] Could not register BidCos-RF_java 
2022-09-25 14:54:23,067 de.eq3.ccu.bidcos.dispatcher.BidCosRpcDispatcher ERROR [Timer-2] Could not register BidCos-RF_java 
2022-09-25 14:54:28,067 de.eq3.ccu.bidcos.dispatcher.BidCosRpcDispatcher ERROR [Timer-2] Could not register BidCos-RF_java 
2022-09-25 14:54:33,064 de.eq3.ccu.bidcos.dispatcher.BidCosRpcDispatcher ERROR [Timer-2] Could not register BidCos-RF_java 
2022-09-25 14:54:38,066 de.eq3.ccu.bidcos.dispatcher.BidCosRpcDispatcher ERROR [Timer-2] Could not register BidCos-RF_java 

Wenn ich trotzdem versuche Geräte einzulernen, sehe ich zwar, dass der Anlernmodus aktiv ist
![image](https://user-images.githubusercontent.com/40394719/192144885-b45be81f-cedb-4ef8-a604-eab7f0bf5e30.png)
aber nachdem ich meine Thermostate resettet habe (Batterien raus, alle 3 Tasten gedrückt halten, Batterien rein) und in den Anlernmodus versetzt habe (3 Sek auf Boost) erscheinen diese nicht in der Inbox. Die Thermostate zählen richtigerweise von 30 runter, wenn der Einlernmodus auf der RaspberryMatic nicht aktiv ist und bestätigen den Einlernprozess mit _AC_ wenn der Einlernmodus der RaspberryMatic aktiv ist. Es scheint also so, als wenn die Thermostate eine Verbindung herstellen können!?

Meine Suche nach ähnlichen Problemen war leider nicht erfolgreich. Ich habe zwar die Diskussion hier verfolgt _https://homematic-forum.de/forum/viewtopic.php?f=65&t=38633_ aber mein HM-MOD-RPI-PCB ist richtig zusammengelötet - er funktionierte immerhin jahrelang mit FHEM gemeinsam. 

Ich bin leider auch nicht der große Experte was Homematic und Home Assistant angeht, trotzdem hoffe ich, dass ich mein Problem ausreichend beschrieben habe. Über jede Hilfe wäre ich sehr dankbar!

Gruß
Jan

**Comment by jens-maus** [comment, marked as answer]
> Starting rfd: ....................ERROR

Das ist dein problem und hängt damit zusammen das es momentan noch seit HomeAssistantOS 9.x eine kleine Inkompatibilität gibt die es notwendig macht entweder auf HomeAssistant 8.x downzugraden oder den aktuellste nightly snapshot von RaspberryMatic in Kombination mit HomeAssistant Supervisor version 2022.09.1 oder neuer zu verwenden (siehe https://github.com/jens-maus/RaspberryMatic/issues/1942). 

Also einfach HomeAssistantOS downgraden oder auf den nächsten RaspberryMatic Release warten oder den aktuellen nightly snapshot verwenden.

**Reply by MrJanD** [reply]
Danke für die schnelle Antwort!!

**Comment by northnorway** [comment]
Ich habe das exakt gleiche Problem und habe heute erfreut festgestellt, dass seit heute ein Update für das Raspberrymatic-Addon zur Verfügung steht. Das Update (3.65.11.20221005) ändert bei mir allerdings nichts an dem Problem. Mache ich etwas falsch oder muss ich mich einfach noch gedulden bis zum nächsten Update? Danke für eine kurze Info hierzu!
